app/utilities/mappings.py

```diff
@@ -35,3 +35,8 @@ def _handle_sequence(value: Sequence) -> Generator[Mapping, None, None]:
                 yield from get_mappings_with_key(key, v, ignore_keys)
             if isinstance(v, Sequence):
                 yield from _handle_sequence(v)
+
+
+def get_values_for_key(key: str, data: Mapping | Sequence) -> Generator:
```

Comment:
Minor question (not related to this PR really, as I think it might have been there before the type hinting updates) - would the `Generator` yield type narrow down to a `Mapping`? E.g. something like `-> Generator[Mapping, None, None]`?

Reply:
In this case its getting all the values out of a `Mapping` rather than the mappings themself like the one above it so for example if data is
```
data = { "store": [1, 2, 3], "test": { "store": 1 }, "nested_1": [{"nested_2": { "store": "xyz" }}] }
```
the get values for key would generate `[1, 2, 3]` then `1` then `"xyz"` so it could be anything and I'm not able to narrow down the type

Reply:
Ah I see, cool that makes sense - thanks 👍